apiserver/tools.go

```diff
@@ -49,12 +51,18 @@ func (h *toolsDownloadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 		tarball, err := h.processGet(r, st)
 		if err != nil {
 			logger.Errorf("GET(%s) failed: %v", r.URL, err)
-			sendError(w, errors.NewBadRequest(err, ""))
+			if err := sendError(w, errors.NewBadRequest(err, "")); err != nil {
```

Comment:
Personally, I agree with Tim here.  If this is always called from an http handler that itself can't return errors, and we always just log the error, then forcing the repeat of the error handling everywhere is not very useful, and makes the code harder to follow, especially since this is already in an error handling path.

We don't ever branch based on the error, other than to log.  Returning it is just not useful.  We're in an error path, it's understood that if there are failures in the error path then you're kind of screwed.  as long as sendError has an appropriate doc comment that explains that it may fail, but then there's nothign we can do about it, so we just log it... then I think it's fine.

Sure, it's usually nice to know when a function can fail, but if you know you'll never actually be able to do anything about it, and it's just an internal function, then I don't think it's worth the extra code to return that function one level and always handle it the same way.

Plus, if the error handling is inside sendError, then you know the error handling is always the same, without having to copy and paste the code.

Finally, it should be really rare that there's a bad request *and* the connection dies at just the wrong nanosecond... so we're wasting visual complexity on something that in practice will almost never happen.